resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+| Open sourcing templates | Repository and pipeline to validate, testing the infrastructure and publishing  | Desire to contribute, easy to use, low overhead /dependency to publish |
+
+## Solution approach in Radius
+
+### Main job : Build a custom platform for application development
+
+#### Define the set of key technologies for the enterprise applications
+Enterprises choose technologies based on the nature of technology, open-source support, documentation support, existing vendor relations, expertise in the team. Radius can provide the following to the users:
+
+*Community supported assets* : Community supported assets include the resource type definitions and IaC templates to deploy and manage resources in applications. Radius can enable the community to provide and use the assets for the prevalent cloud native technologies. This will help users to use pre-existing assets IaC templates and definitions in their applications without having to author them from scratch. 
+
+*Community supported assets meets their requirements* : Enterprises have a wide range of requirements for their applications based on the nature of the application, market, region, hosting models,compliance and security policies. Radius can enable the community to provide and use the resource type definitions and IaC templates that meet the requirements of the enterprises and meet them where they are. This will enable users to build what they want via Radius thus unblocking them to build their applications.
+
+*Maintenance and support for the community supported assets* : Radius can provide the necessary tools and services to maintain and support the community supported assets. This will help the community to contribute, maintain the assets and provide support to the users who are using the assets.
+
+**Solution in Radius**
+
+1. Resource extensibility for defining the custom resource types in Radius
+1. Recipes for the custom resource types
+1. Community supported resource type definitions
+1. Community supported Recipes
+1. Radius auto generates AWS, Azure Recipes from the definitions (Recipes for any resource) 
+
+We will cover the design and user experience for all the above solutions except for "Recipes for any resource" in this document to keep the scope of the work simple.
+
+#### Specify the contract for the key services in the application platform
+
+As a platform engineer, I want to provide the custom platform with the following: 
+
+*Resource schema authoring*: Define a schema for the custom resource types that my application uses in Radius. The schema defines the structure and properties of the custom resource type. 
+
+*Auto guidance for schema authoring*: The resource schema authoring tool provides auto guidance and validation to help the user author the schema. The auto guidance tool provides suggestions and recommendations to the user based on the schema that they are authoring. 
```

Comment:
Would like to hear more about auto-guidance .. is this copilot? is this intellisense? something else?

Reply:
change this to auto completion for clarity